scripts/dts/dtlib.py

```diff
@@ -1354,6 +1347,32 @@ class Property:
 
       See the to_*() methods for converting the value to other types.
 
+    type:
+      The type of the property, inferred from the syntax used in the
+      assignment. This is one of the following constants (with example
+      assignments):
+
+        Assignment         | Property.type
+        -------------------+------------------------
+        foo;               | dtlib.TYPE_EMPTY
+        foo = []           | dtlib.TYPE_BYTES
+        foo = [01 02]      | dtlib.TYPE_BYTES
+        foo = /bits/ 8 <1> | dtlib.TYPE_BYTES
+        foo = <1>          | dtlib.TYPE_NUM
+        foo = <>           | dtlib.TYPE_NUMS
+        foo = <1 2 3>      | dtlib.TYPE_NUMS
+        foo = <1 2>, <3>   | dtlib.TYPE_NUMS
+        foo = "foo"        | dtlib.TYPE_STRING
+        foo = "foo", "bar" | dtlib.TYPE_STRINGS
+        foo = <&label>     | dtlib.TYPE_PHANDLE
```

Comment:
```
foo = <&foo, &bar>     | dtlib.TYPE_PHANDLES 
```
is missing to check
```
pinctrl-0 = <&foo, &bar>
```

Reply:
I'm planning to add `type: phandle-array` as well. The PR is a bit bloated already through, so thought it could be done separately.

Reply:
Thinking the condition for `phandle-array` could be "one or more phandles and 32-bit data values", to allow it to match `<&foo 1 2 &bar 3 4>` as well.

Reply:
Hmm... maybe there could be a `phandles` type for pure-phandle arrays, and a `phandle-array` for cases where you can also mix in data.

That'd be a step up from just having `phandle-array`.

Reply:
PHANDLES lgtm for a pure phandles list. Detection by all values start with &. In edtlib the <XXX>S specifiers from dtlib are mapped to <xxx>-array (e.g. STRINGS -> string-array). Keeping this convention edtlib should use phandle-array for the pure list of phandles.

Maybe the mixed phandle & int list can become PHCOMPOUND for a single compound and PHCOMPOUNDS for a list of. Detection though is somewhat tricky as it may be just a mixed value compound. Binding type in edtlib could be phandle-compound and phandle-compound-array.